### ISSUE TITLE
Search for open affixes

### ISSUE THREAD

**Issue by hexedsilicon** [thirdy/durian#57]
Is it possible to search to see if there is an open affix? It'd be useful for crafting if I could, for example, search "ring 50life 40eleres 1opensuffix" or something similar.

**Comment by kyri0** [comment]
Durian can only search what poe.trade can search, and as far as my aware, you can't do that kind o search on poe.trade, but yes, it would be indeed a huge help. I don't even know if Exile Tools can do it, because this a kind of search that would require the program to know all the mods on the item - like Durian does to show if a mod is a suffix or prefix, and its tier - and filter the search by them. For now, you will have to look it for yourself, but certainly Durian helps a lot in that regard.

**Comment by hexedsilicon** [comment]
It might not be possible for poe.trade, but isn't durian getting a list? It might be possible to only display the ones that have x number of suffix/prefix?

**Comment by Dominik-O22** [comment]
but as you said kyri0, poe.trade and durian know every mod that is on them item, and durian knows for every mod if its a suffix or a prefix, so durian should be able to count them, and if you can count them you can also do a simple check against the max amount of suffixes/prefixes

**Comment by thirdy** [comment]
@zocke1r that can be easily done. note that poe.trade only returns 99 items. and out of those items, the filtering of items with an open suffix can be done.

also note that hybrid mods are ignored at the moment. so items with those mods will be tackled later. and yeah, there are cases where you cannot really tell  if a hybrid mod has taken two affixes or one so those are out of the question.

we'll consider this ticket as the ticket for that implementation.

**Comment by kyri0** [comment]
@zocke1r 
Yes, I thought to mention that it could probably be done at Durian's end (sadly without 100% accuracy cuz of hybrids and legacy mods), but as I couldn't be sure, I just explained how Durian basically works, which is sending poe.trade requests and that it can only search what poe.trade does.

The legacy mods we can probably easily be fixed, and they aren't many. For the hybrids, we can try to do something like PoE Item Info Script does (although it also doesn't work 100%). Can you have a look at it, @thirdy ? [https://www.pathofexile.com/forum/view-thread/1463814](url)

**Comment by thirdy** [comment]
Last time I looked into the item info script, it was a bit complicated (for me), mostly due to AHK not really meant for this kind of programming as well as my unfamiliarity of it.

but really, it uses text files to store the data on mods and parses them manually. vs what we did which is all nicely contained in one big json object.

ultimately, these are the steps that need to be achieved:
1. Gather examples of items where hybrid mods can be identified.
2. Gather examples of items where hybrid mod is impossible to identify.
3. Lay down the steps on how to re-construct the original mod.

But let's wait for the new GGG API, they might give us this :dancer:
